examples/chart/teleport-kube-agent/tests/statefulset_test.yaml

```diff
@@ -44,6 +44,20 @@ tests:
       - matchSnapshot:
           path: spec.template.spec
 
+  - it: sets StatefulSet annotations when specified
+    template: deployment.yaml
+    values:
+      - ../.lint/stateful.yaml
+      - ../.lint/annotations.yaml
+    asserts:
+      - equal:
+          path: metadata.annotations.kubernetes\.io/deployment
+          value: test-annotation
+      - equal:
+          path: metadata.annotations.kubernetes\.io/deployment-different
+          value: 3
+      - matchSnapshot: {}
```

Comment:
```suggestion
```

Nit: the snapshot test is overkill, I know we already have a bunch of them all over the place but they don't add much.

Reply:
i've removed it